src/maker.rs

```diff
@@ -51,13 +51,34 @@ impl Maker {
             dai_reserved_funds: Default::default(),
             btc_max_sell_amount,
             dai_max_sell_amount,
-            mid_market_rate,
+            mid_market_rate: Some(mid_market_rate),
             spread,
             ongoing_takers: Default::default(),
         }
     }
 
-    pub fn update_rate(&mut self, mid_market_rate: ()) {}
+    pub fn update_rate(
+        &mut self,
+        mid_market_rate: MidMarketRate,
+    ) -> anyhow::Result<RateUpdateDecision> {
```

Comment:
```
	) -> anyhow::Result<Option((Order, Order))> {
```
I don't think the `RateUpdateDecision` enum adds much value. It's unneeded complexity, just return an option and if the option contain orders, publish them.

Reply:
I only kept that enum because an option of a tuple struct - `Option<(BtcDaiOrder, BtcDaiOrder)>` -would imply the sell and buy order arbitrarily and I did not like that.

That can also be solved by wrapping the orders by another struct - I pushed this as a change for now, but I feel the complexity is more or less the same. 

Reducing the output to just an `Option` would imply that we don't bubble up the `Result` of `new_xxx_order()`.